### PR TITLE
Respect minDate and maxDate in Dropdown Scroll Mode

### LINKED ISSUE #816: Year Dropdown Scroll Mode Ignores minDate and maxDate
in year_dropdown.jsx
Select mode does not ignore these limits, but scroll mode does.

If I get a chance I will try and fix this myself.

### PR DESCRIPTION
**Pull request by mgrdevport**
Fixes #816 
* Limit scrollable year dropdown to years between max and min date, if provided.
* Hide icons to add previous/upcoming year, if year of max/min date is already included in scrollable dropdown list.
* Set max-height to scrollable dropdown, to allow it to shrink down if neccessary.

### CODE REVIEW

**Comment by martijnrusschen** [comment]
Nice work, I just merged a PR that changes the codebase to ES6 classes that's causing merge conflicts right now. I can merge this after the conflicts are resolved.

File: src/year_dropdown_options.jsx [hunk 1]
```diff
@@ -1,10 +1,17 @@
 import React from 'react'
 import classNames from 'classnames'
 
-function generateYears (year, noOfYear) {
+function generateYears (year, noOfYear, minDate, maxDate) {
   var list = []
   for (var i = 0; i < (2 * noOfYear); i++) {
-    list.push(year + noOfYear - i)
+    const newYear = year + noOfYear - i
+    if (minDate && maxDate) {
```
Comment (aij) [inline src/year_dropdown_options.jsx]: Shouldn't this honor `minDate` and `maxDate` even when only one of the two is specified?

Comment (mgrdevport) [inline src/year_dropdown_options.jsx]: @aij Good point, I will work on this later.

**Comment by vuphuctho** [comment]
Hi what is the current status of this PR?

File: src/year_dropdown_options.jsx [hunk 2]
```diff
@@ -15,19 +25,29 @@ export default class YearDropdownOptions extends React.Component {
...
 
   constructor (props) {
     super(props)
     this.state = {
-      yearsList: this.props.scrollableYearDropdown ? generateYears(this.props.year, 10) : generateYears(this.props.year, 5)
+      yearsList: this.getYearsList(this.props.year, this.props.minDate, this.props.maxDate)
+    }
+  }
+
+  getYearsList = (year, minDate, maxDate) => {
```
Comment (aij) [inline src/year_dropdown_options.jsx]: These function arguments are not used.

File: docs-site/bundle.js [hunk 3]
```diff
@@ -65,14 +65,14 @@
 	return modules;
 }([
 /* 0 */
-/***/ function(module, exports, __webpack_require__) {
+/***/ (function(module, exports, __webpack_require__) {
```
Comment (aij) [inline docs-site/bundle.js]: Please avoid committing build artifacts, as they produce unnecessary conflicts and noise.

File: src/stylesheets/datepicker.scss [hunk 4]
```diff
@@ -250,7 +250,7 @@
   }
 
   &--scrollable {
-    height: 150px;
+    max-height: 150px;
```
Comment (aij) [inline src/stylesheets/datepicker.scss]: Please avoid committing build artifacts, as they produce unnecessary conflicts and noise.

File: src/year_dropdown_options.jsx [hunk 5]
```diff
@@ -37,22 +57,26 @@ export default class YearDropdownOptions extends React.Component {
...
-      <div className="react-datepicker__year-option"
-          ref={'previous'}
-          key={'previous'}
-          onClick={this.decrementYears}>
-        <a className="react-datepicker__navigation react-datepicker__navigation--years react-datepicker__navigation--years-previous" />
-      </div>
-    )
+    if (!this.props.maxDate || !this.state.yearsList.find(year => year === this.props.maxDate.year())) {
+      options.unshift(
+              <div className="react-datepicker__year-option"
```
Comment (aij) [inline src/year_dropdown_options.jsx]: Please follow the existing indentation style. (Two spaces, and closing parenthesis aligned with the start of the line it was opened on.)

Sorry, we should probably automate this...

File: test/year_dropdown_options_test.js [hunk 6]
```diff
@@ -14,11 +15,11 @@ describe('YearDropdownOptions', () => {
     sandbox = sinon.sandbox.create()
     onCancelSpy = sandbox.spy()
     yearDropdown = mount(
-      <YearDropdownOptions
-          year={2015}
-          onChange={mockHandleChange}
-          onCancel={onCancelSpy} />
-    )
+            <YearDropdownOptions
```
Comment (aij) [inline test/year_dropdown_options_test.js]: More (accidental?) indentation... Also below.

**Comment by mgrdevport** [comment]
I will do this on the next weekend, sorry for my late reply.

**Comment by pjjanak** [comment]
Any movement on this?

**Comment by martijnrusschen** [comment]
This got fixed by #1072.

**Pull request closed without merging**